File: system-test/dotnet/SystemTests/Legacy/Mod10/AcceptanceTests/PlaceOrderPositiveIsolatedTest.cs
using SystemTests.Legacy.Mod10.AcceptanceTests.Base;
using Dsl.Core.Shop;
using Optivem.Testing;
using Xunit;

namespace SystemTests.Legacy.Mod10.AcceptanceTests;

[Collection("Isolated")]
public class PlaceOrderPositiveIsolatedTest : BaseAcceptanceTest
{
    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldApplyFullPriceOnWeekday(Channel channel)
    {
        await Scenario(channel)
            .Given().Product().WithUnitPrice(20.00m)
            .And().Clock().WithWeekday()
            .When().PlaceOrder().WithQuantity(5)
            .Then().ShouldSucceed()
            .And().Order()
                .HasTotalPrice(100.00m);
    }

    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldApplyWeekendDiscount(Channel channel)
    {
        await Scenario(channel)
            .Given().Product().WithUnitPrice(20.00m)
            .And().Clock().WithWeekend()
            .When().PlaceOrder().WithQuantity(5)
            .Then().ShouldSucceed()
            .And().Order()
                .HasTotalPrice(50.00m);
    }

    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldRecordPlacementTimestamp(Channel channel)
    {
        (await Scenario(channel)
            .Given().Clock()
                .WithTime("2026-01-15T10:30:00Z")
            .When().PlaceOrder()
            .Then().ShouldSucceed()
            .And().Clock())
            .HasTime("2026-01-15T10:30:00Z");
    }
}

File: system-test/java/src/test/java/com/optivem/eshop/systemtest/legacy/mod10/acceptance/PlaceOrderPositiveIsolatedTest.java
package com.optivem.eshop.systemtest.legacy.mod10.acceptance;

import com.optivem.eshop.systemtest.legacy.mod10.acceptance.base.BaseAcceptanceTest;
import com.optivem.eshop.dsl.channel.ChannelType;
import com.optivem.testing.Channel;
import com.optivem.testing.Isolated;

import org.junit.jupiter.api.TestTemplate;

@Isolated
class PlaceOrderPositiveIsolatedTest extends BaseAcceptanceTest {
    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldApplyFullPriceOnWeekday() {
        scenario
                .given().product().withUnitPrice(20.00)
                .and().clock().withWeekday()
                .when().placeOrder().withQuantity(5)
                .then().shouldSucceed()
                .and().order()
                    .hasTotalPrice(100.00);
    }

    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldApplyWeekendDiscount() {
        scenario
                .given().product().withUnitPrice(20.00)
                .and().clock().withWeekend()
                .when().placeOrder().withQuantity(5)
                .then().shouldSucceed()
                .and().order()
                    .hasTotalPrice(50.00);
    }
}
